cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkPackageUtils.java

```diff
@@ -493,7 +493,12 @@ private static SparkFramework prepareSparkFramework(SparkCompat sparkCompat,
 
   private static boolean uploadToLocation(Location frameworkLocation, File archive, Location writeLockLocation)
     throws IOException {
-    if (!writeLockLocation.createNew()) {
+    try {
+      if (!writeLockLocation.createNew()) {
+        return false;
+      }
+    } catch (IOException e) {
+      LOG.warn("Failed to upload spark framework artifact while creating lock file and will be retried.", e);
```

Comment:
Since it's an expected exception when there is concurrent creation of the file, you can log it as debug